vm/src/obj/objdict.rs

```diff
@@ -313,25 +311,28 @@ macro_rules! dict_iterator {
         #[derive(Debug)]
         struct $iter_name {
             pub dict: PyDictRef,
-            pub position: Cell<usize>,
+            pub position: RefCell<dictdatatype::DictPosition>,
```

Comment:
If you made `DictPosition` `Copy`, you could keep this as a `Cell`.

Reply:
Actually, does this even need to be mutable? It looks like you're not changing it in `__next__`.

Reply:
Thinking more on this, given the eventual implementation I use position and size in different places so there's no need to bundle them together.